src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs

```diff
@@ -55,6 +56,12 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
 
             foreach (var resourceContainer in context.Library.ResourceContainers)
             {
+                // Singleton does not have container class
+                if (resourceContainer.OperationGroup.IsSingletonResource(context.Configuration.MgmtConfiguration))
```

Comment:
Can we move this to the output library?  So the output library only creates the ResourceContainers that should exist instead of skipping at generation time?